OpenActive.Server.NET/OpenBookingHelper/Async/AsyncDuplicateLock.cs

```diff
@@ -0,0 +1,67 @@
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace OpenActive.Server.NET.OpenBookingHelper.Async
+{
+    public sealed class AsyncDuplicateLock
```

Comment:
Could the `AsyncDuplicateLock` be generic, and the `object` be typed? So e.g. it could work for both a string and a GUID? Perhaps making this slightly more robust to use?

Reply:
Yep, good shout.  Have genericised it...